fe/src/main/java/org/apache/doris/resource/Tag.java

```diff
@@ -0,0 +1,113 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.resource;
+
+import org.apache.doris.common.AnalysisException;
+import org.apache.doris.common.io.Text;
+import org.apache.doris.common.io.Writable;
+import org.apache.doris.common.util.GsonUtils;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableSet;
+import com.google.gson.annotations.SerializedName;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Objects;
+
+/*
+ * A Tag consists of type and name.
+ * Tag type and name are both case insensitive, and represented in lower case.
+ * Tag is printed as { "type": "name" }
+ * 
+ * A Tag is immutable once it being created.
+ */
+public class Tag implements Writable {
+    // Classification of tag
+    public enum Type {
```

Comment:
I think we can make this type as a string. In this way, all other tag will be classified into CUSTOM which is not convenient for developer to collect his interesting tag. If we make it a string, new tags can be classified to new type according its usage, it will be easy for developer to collect their interesting tags.

Reply:
OK, I will change it String